app/views/posts/index.html.erb

```diff
@@ -25,14 +26,17 @@
             </li>
           <% end %>
         </ul>
+
+        <% if can?(:destroy, post) %>
+          <%= button_to "Delete Post", user_post_path(@user, post), method: :delete, data: { confirm: "Are you sure?" }, class: "btn btn-danger" %>
+        <% end %>
```

Comment:
**_I had to create a post from the rails console to check other requirements._**

- [ ] Kindly look into your app so that it works as required. From the project guidelines, only the admin and the user who has the post should be able to see the delete button, but all users can see this button in your app. Kindly resolve this. Ensure that only the admin and the user who has the post can see the button, and only these two can delete the post.

- [ ] **[OPTIONAL]** I would recommend you have this delete button in the show page file, not the index file as I think, based on context, it belongs there more than on the index file.